client/sign_in_test.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"encoding/json"
+	"net/url"
+	"testing"
+
+	"github.com/agiledragon/gomonkey/v2"
+	"github.com/logto-io/go/core"
+)
+
+func TestSignInShouldReturnSignInUriCorrectly(t *testing.T) {
+	testAppId := "appId"
+	testAppSecret := "appSecret"
+	testAuthEndpoint := "https://example.com/auth"
+	testRedirectUri := "https://myapp.com/sign-in-callback"
+	testState := "testState"
+	testCodeVerifier := "testCodeVerifier"
+	testCodeChallenge := "testCodeChallenge"
+
+	patchesForGenerateState := gomonkey.ApplyFunc(core.GenerateState, func() string {
+		return testState
+	})
+	defer patchesForGenerateState.Reset()
+
+	patchesForGenerateCodeVerifier := gomonkey.ApplyFunc(core.GenerateCodeVerifier, func() string {
+		return testCodeVerifier
+	})
+	defer patchesForGenerateCodeVerifier.Reset()
+
+	patchesForGenerateCodeChallenge := gomonkey.ApplyFunc(core.GenerateCodeChallenge, func(codeVerifier string) string {
+		return testCodeChallenge
+	})
+	defer patchesForGenerateCodeChallenge.Reset()
+
+	patchesForGenerateSignInUri := gomonkey.ApplyFunc(core.GenerateSignInUri, func(option *core.SignInUriGenerationOptions) (string, error) {
+		return testAuthEndpoint + "?redirect_uri=" + testRedirectUri + "&state=" + testState + "&code_challenge=" + testCodeChallenge, nil
```

Comment:
can we just return "generated_sign_in_uri" here? for unit test, we don't need the real logic.

Reply:
cool, updated.